Sources/ArcGISToolkit/Components/Augmented Reality/WorldScaleSceneView.swift

```diff
@@ -0,0 +1,274 @@
+            Task { await locationDataSource.stop() }
+        }
+        .task {
+            // Start the location data source when the view appears.
+            
+            // Request when-in-use location authorization.
+            // The view utilizes a location datasource and it will not start until authorized.
+            let locationManager = CLLocationManager()
+            if locationManager.authorizationStatus == .notDetermined {
+                locationManager.requestWhenInUseAuthorization()
+            }
+            if !checkTrackingCapabilities(locationManager) {
+                print("Device doesn't support full accuracy location.")
+            }
+            
+            do {
+                try await locationDataSource.start()
+            } catch {
+                self.error = error
+            }
+            for await location in locationDataSource.locations {
+                currentLocation = location
+            }
+        }
+        .task {
+            do {
+                geoTrackingIsAvailable = try await checkGeoTrackingAvailability()
+            } catch {
+                self.error = error
+            }
+        }
+        .overlay(alignment: calibrationViewAlignment) {
+            if !calibrationViewIsHidden {
+                if !isCalibrating {
+                    Button {
+                        withAnimation {
+                            isCalibrating = true
+                        }
+                    } label: {
+                        Text(calibrateLabel)
+                            .padding()
+                    }
+                    .background(.regularMaterial)
+                    .clipShape(RoundedRectangle(cornerRadius: 10))
+                    .disabled(!initialCameraIsSet)
+                    .padding()
+                    .padding(.vertical)
+                }
+            }
+        }
+        .overlay(alignment: .bottom) {
+            if isCalibrating {
+                CalibrationView(viewModel: calibrationViewModel, isPresented: $isCalibrating)
+                    .padding(.bottom)
+            }
+        }
+        .overlay(alignment: .top) {
+            accuracyView
+        }
+    }
+    
+    /// A view that displays the horizontal and vertical accuracy of the current location datasource location.
+    @ViewBuilder
+    var accuracyView: some View {
+        if let currentLocation {
+            VStack {
+                Text("H. Accuracy: \(currentLocation.horizontalAccuracy.formatted(.number.precision(.fractionLength(2))))")
+                Text("V. Accuracy: \(currentLocation.verticalAccuracy.formatted(.number.precision(.fractionLength(2))))")
+            }
+            .multilineTextAlignment(.center)
+            .padding(8)
+            .frame(maxWidth: .infinity, alignment: .center)
+            .background(.regularMaterial)
+            .clipShape(RoundedRectangle(cornerRadius: 10))
+            .padding([.horizontal, .top])
+        }
+    }
```

Comment:
I don't think the location datasource is used other than for this, so it would be nice to clean this up and put it at the app level if it's helpful for debugging.

Reply:
OK, we can take another look when we tackle the "status view" issue, #4515.